docs/INTERNALS.md

```diff
@@ -0,0 +1,328 @@
+# High-Level Design
+![High-Level Design](img/engines_high_level_design.png)
+
+On the diagram above is shown interaction of major components of JerryScript: Parser and Virtual Machine (VM). Parser performs translation of input ECMAScript application into the byte-code with the specified format (refer to [Bytecode](#byte-code) and [Parser](#parser) page for details). Prepared bytecode is executed by the Virtual Machine that performs interpretation (refer to [Virtual Machine](#virtual-machine) and [ECMA](#ecma) pages for details).
```

Comment:
The first sentence is a little bit funky. Maybe something like `The diagram above shows interactions between major components...`?